version: '3'
networks:   
  default:

services:
  back-python:
    build:
      context: ./
      dockerfile: Dockerfile.python.back
    container_name: back-python
    tty: true
    restart: always
    networks:
      - default
    ports:
      - 8080:8080
    healthcheck:
      test: exit $$( curl -so /dev/null --fail http://127.0.0.1:8080 && echo 0 || echo 1 )
      interval: 30s
      timeout: 1s
      retries: 1
    volumes:
      - ./back/python:/home/api:ro
  
  back-python:
    build:
      context: ./
      dockerfile: Dockerfile.node.back
    container_name: back-node
    tty: true
    restart: always
    networks:
      - default
    ports:
      - 8081:8080
    healthcheck:
      test: exit $$( curl -so /dev/null --fail http://127.0.0.1:8080 && echo 0 || echo 1 )
      interval: 30s
      timeout: 1s
      retries: 1
    volumes:
      - ./back/python:/home/api:ro


  rethink:
    image: rethinkdb:latest
    container_name: rethink
    restart: always
    networks:
      - default
    ports:
      - 8082:8080
    expose:
      - 28015
    volumes:
      - ./back/rethink/data:/data
    command: rethinkdb --bind all -d /data
    
